public/app/plugins/datasource/loki/components/monaco-query-field/monaco-completion-provider/CompletionDataProvider.ts

```diff
@@ -8,11 +8,12 @@ import { LokiQuery } from '../../../types';
 
 import { Label } from './situation';
 
+interface HistoryRef {
+  current: Array<HistoryItem<LokiQuery>>;
+}
+
 export class CompletionDataProvider {
-  private history: string[] = [];
-  constructor(private languageProvider: LanguageProvider, history: Array<HistoryItem<LokiQuery>> = []) {
-    this.setHistory(history);
-  }
+  constructor(private languageProvider: LanguageProvider, private historyRef: HistoryRef = { current: [] }) {}
```

Comment:
do we need the default-value for the second param? i did a quick check and it does not seem that we are calling this constructor without the second param anywhere.

Reply:
No. It's just a safety measure because history has been undefined in the past.